rustls/src/client/hs.rs

```diff
@@ -77,21 +77,22 @@ fn find_session(
     sess: &mut ClientSessionImpl,
     dns_name: webpki::DNSNameRef,
 ) -> Option<persist::ClientSessionValue> {
-    let key = persist::ClientSessionKey::session_for_dns_name(dns_name);
-    let key_buf = key.get_encoding();
-
-    let maybe_value = sess
-        .config
-        .session_persistence
-        .get(&key_buf);
+    let key_buf = {
+        let key = persist::ClientSessionKey::session_for_dns_name(dns_name);
+        key.get_encoding()
+    };
 
-    if maybe_value.is_none() {
-        debug!("No cached session for {:?}", dns_name);
-        return None;
-    }
+    let config = &sess.config.session_persistence;
+  
+    let value = match config.get(&key_buf) {
+        Some(v) => v,
+        None => {
+            debug!("No cached session for {:?}", dns_name);
+            return None;
+        }
+    };
 
-    let value = maybe_value.unwrap();
-    let mut reader = Reader::init(&value[..]);
```

Comment:
switching from `&value[..]` to `value.as_slice()` is also an out-of-scope change that should be reverted.

If we want to switch the coding pattern here, we should probably do so globally. Probably we should just do what Clippy recommends. Note that `as_slice()` is also unneeded here; just `&value` is enough. But, let's save that for another day.

Reply:
Reverted for now as well.